src/web/handlers/contacts.rs

```diff
@@ -1,64 +1,65 @@
 use super::super::data::{DeleteContactForm, EditContactForm, NewContactForm};
-use crate::bill::contacts::{
-    add_in_contacts_map, change_contact_name_from_contacts_map, delete_from_contacts_map,
-    get_contacts_vec, Contact,
-};
 use crate::constants::IDENTITY_FILE_PATH;
-use crate::dht::Client;
+use crate::service::contact_service::Contact;
+use crate::service::{self, Result, ServiceContext};
 use rocket::form::Form;
 use rocket::http::Status;
 use rocket::serde::json::Json;
 use rocket::{delete, get, post, put, State};
 use std::path::Path;
 
 #[get("/return")]
-pub async fn return_contacts() -> Json<Vec<Contact>> {
-    let contacts: Vec<Contact> = get_contacts_vec();
-    Json(contacts)
+pub async fn return_contacts(state: &State<ServiceContext>) -> Result<Json<Vec<Contact>>> {
+    let contacts: Vec<Contact> = state.contact_service.get_contacts().await?;
+    Ok(Json(contacts))
 }
 
 #[delete("/remove", data = "<remove_contact_form>")]
-pub async fn remove_contact(remove_contact_form: Form<DeleteContactForm>) -> Status {
+pub async fn remove_contact(
+    remove_contact_form: Form<DeleteContactForm>,
+    state: &State<ServiceContext>,
+) -> Result<Status> {
     if !Path::new(IDENTITY_FILE_PATH).exists() {
```

Comment:
Not really related to this PR, but what's the point of this precondition check? :thinking:

I guess we're checking if the app was initialized (but we're creating the folders before we start the server, so in that case we wouldn't need it), or make sure that the `identity` file hasn't been deleted, rendering the receiving node somewhat invalid.

But then one could just write gibberish into the `./identity/identity` file, which would have the same effect.

In any case, if we have this precondition (if it's still valid) on every handler, then we could put it into a middleware in a separate task (to make sure we don't forget to add it). What do you think?

(cc @mtbitcr)

Reply:
Yes for those i did not want to change to much of the actual logic before having tests available and this pattern is used in other places. With current impl. I could test out the early return of error status codes so this should be easy to add in other places we want to use Result in as well.

Reply:
@mtbitcr and I discussed this shortly today. The idea is to have some check, if the identity is actually initialized and if not, that all non-identity endpoints return Not-Acceptable.

Created a backlog task for this here: https://github.com/BitcreditProtocol/E-Bill/issues/193